python/BioSimSpace/Sandpit/Exscientia/Process/_openmm.py

```diff
@@ -1949,6 +1920,49 @@ def _add_config_platform(self):
                 "properties = {'OpenCLDeviceIndex': '%s'}" % opencl_devices
             )
 
+    def _add_config_restraints(self):
+
+        # Add backbone restraints. This uses the approach from:
+        # https://github.com/openmm/openmm/issues/2262#issuecomment-464157489
+        # Here zero-mass dummy atoms are bonded to the restrained atoms to avoid
+        # issues with position rescaling during barostat updates.
+        restraint = self._protocol.getRestraint()
+        if restraint is not None:
+            # Search for the atoms to restrain by keyword.
+            if isinstance(restraint, str):
+                restrained_atoms = self._system.getRestraintAtoms(restraint)
+            # Use the user-defined list of indices.
+            else:
+                restrained_atoms = restraint
+
+            # Get the force constant in units of kJ_per_mol/nanometer**2
+            force_constant = self._protocol.getForceConstant()._sire_unit
+            force_constant = force_constant.to(
+                _SireUnits.kJ_per_mol / _SireUnits.nanometer2
+            )
+
+            self.addToConfig(
+                "\n# Restrain the position of atoms using zero-mass dummy atoms."
+            )
+            self.addToConfig("restraint = HarmonicBondForce()")
+            self.addToConfig("restraint.setUsesPeriodicBoundaryConditions(True)")
+            self.addToConfig("system.addForce(restraint)")
+            self.addToConfig(
+                "nonbonded = [f for f in system.getForces() if isinstance(f, NonbondedForce)][0]"
+            )
+            self.addToConfig("dummy_indices = []")
+            self.addToConfig("positions = inpcrd.positions")
+            self.addToConfig(f"restrained_atoms = {restrained_atoms}")
```

Comment:
You might want to have a unit test to check is `"restrained_atoms = [1, 2, 3, 4]"` is in the script.

Reply:
But this is the purpose of the regression test. Instead of checking for individual lines in the file, the whole generated file is checked against a validated version of this file, where the restraints code is as we expect it. The test fails if the generated file and the reference file differ.